src/common/template.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import fs from "fs-extra";
+import Handlebars from "handlebars";
+
+/**
+ * Holds a template and renders the template with the given data
+ */
+export class Template {
+  content: string;
+
+  /**
+   * Creates a Template object
+   * @param path - Path to the template file
+   * @param handlebars - Optional handlebars instance, default instance is used if not provided
+   */
+  constructor(public path: string, public handlebars = Handlebars) {
+    try {
+      this.content = fs.readFileSync(path, "utf8");
+    } catch (error) {
+      error.message = `Error initializing template: ${error.message}`;
+      throw error;
+    }
+    if (!this.content) {
+      throw new Error(`Invalid template content: ${this.content}`);
+    }
+  }
+
+  /**
+   * Make substitutions and write the template to a file.
+   *
+   * @param context - The data model to use for substitutions
+   * @param destination - The file path to write the rendered template to
+   */
+  render(context: { [key: string]: unknown }, destination: string): void {
```

Comment:
Should this be returning void?

Reply:
yes, it renders to the given destination.

Reply:
Is it synchronous?

Reply:
(And should it be synchronous if it is?)

Reply:
its synchronous right now

Reply:
Should it be synchronous?

Reply:
keeping it synchronous lets us handle any errors better (I added try catch around it)

Reply:
The error handling is the same for sync/async. All you'd need to change here is `fs.outputFileSync` to `await fs.outputFile` and the return type to `Promise<void>`.